packages/nextcloud_test/lib/src/test_target/local.dart

```diff
@@ -0,0 +1,114 @@
+import 'dart:async';
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:nextcloud_test/src/presets.dart';
+import 'package:nextcloud_test/src/test_target/test_target.dart';
+import 'package:process_run/process_run.dart';
+import 'package:version/version.dart';
+
+/// Factory for running tests against a local instance.
+class LocalFactory implements TestTargetFactory<LocalInstance> {
+  @override
+  LocalInstance spawn(Preset? preset) => LocalInstance();
+
+  @override
+  Map<String, List<Version>> getPresets() {
+    final presets = <String, List<Version>>{};
+    final regex = RegExp('  - (.*): (.*)');
+
+    final dir = Platform.environment['DIR']!;
+
+    var result = runExecutableArgumentsSync(
+      'php',
+      [
+        '-f',
+        './occ',
+        'app:list',
+        '--enabled',
+      ],
+      workingDirectory: dir,
+    );
+    if (result.exitCode != 0) {
+      throw Exception('Failed to list apps\n${result.stderr}\n${result.stdout}');
+    }
+
+    for (final line in (result.stdout as String).split('\n')) {
+      final matches = regex.allMatches(line);
+
+      if (matches.isNotEmpty) {
+        final match = matches.single;
+
+        presets[match.group(1)!] = [Version.parse(match.group(2)!)];
+      }
+    }
+
+    result = runExecutableArgumentsSync(
+      'php',
+      [
+        '-f',
+        './occ',
+        'status',
+      ],
+      workingDirectory: dir,
+    );
+    if (result.exitCode != 0) {
+      throw Exception('Failed to get status\n${result.stderr}\n${result.stdout}');
+    }
+
+    for (final line in (result.stdout as String).split('\n')) {
+      final matches = regex.allMatches(line);
+
+      if (matches.isNotEmpty) {
+        final match = matches.single;
+        if (match.group(1)! != 'version') {
+          continue;
+        }
+
+        presets['server'] = [Version.parse(match.group(2)!)];
+        break;
+      }
+    }
+
+    return presets;
+  }
+}
+
+/// Test target representing a local instance.
+class LocalInstance extends TestTargetInstance {
+  @override
+  void destroy() {}
+
+  @override
+  Uri hostURL = Uri.parse(Platform.environment['URL']!);
+
+  @override
+  late Uri targetURL = hostURL;
+
+  @override
+  FutureOr<String> createAppPassword(String username) async {
```

Comment:
This always returns a future
```suggestion
  Future<String> createAppPassword(String username) async {
```